src/libbson/src/bson/bson-value.c

```diff
@@ -35,6 +35,7 @@ bson_value_copy (const bson_value_t *src, /* IN */
       dst->value.v_double = src->value.v_double;
       break;
    case BSON_TYPE_UTF8:
+      BSON_ASSERT (src->value.v_utf8.len <= UINT32_MAX - 1);
```

Comment:
`bson_malloc` accepts a `size_t`, so the resulting value is already cast to a `size_t`. I suggest casting to `size_t` first:

```c
BSON_ASSERT (bson_in_range_size_t_unsigned (src->value.v_utf8.len));
size_t len_sz = (size_t) src->value.v_utf8.len;
BSON_ASSERT (len_sz <= SIZE_MAX - 1);
dst->value.v_utf8.len = src->value.v_utf8.len;
dst->value.v_utf8.str = bson_malloc (len_sz + 1);
```

On most 64-bit systems, I expect `size_t` has range greater than `uint32_t`. That may make a failed assert less likely.

Suggestion requires `#include <bson/bson-cmp.h>` for `bson_in_range_size_t_unsigned`.

Reply:
Completely missed that `bson_malloc` takes in a `size_t`. This is definitely a better solution to prevent failed asserts.